### ISSUE TITLE
Full Improved Point of Sale Checkout Flow

### ISSUE THREAD

**Issue by dstrukt** [btcpayserver/btcpayserver#4114]
This issue will track and highlight an end-to-end flow of the improved POS (specifically, but could be any plug-in, etc..) checkout experience, so we evaluate it's strengths and weaknesses. I will create sub issues to break out the relevant tickets to close as we merge, however this ticket should not be closed until all of the following is done.

Note: There are areas of this that are highly WIP (store settings, form builder, cart) that will be determined by the implementation and constraints, but I wanted to start laying out the pieces so we can have more concrete discussions of trade-offs, expectations, etc..

- [x] Whitelabel support + store icon (https://github.com/btcpayserver/btcpayserver/pull/4131)
- [x] Extend whitelabel support (https://github.com/btcpayserver/btcpayserver/issues/3842)
- [x] Improved file system support
- [x] Dynamic form builder (+ eventual plug-in support)
- [x] Updated checkout invoice
- [x] Updated receipt view
- [ ] Updated Point of Sale + Cart view
- [ ] POS: Make it clearer, how the custim amount works ([details](https://github.com/btcpayserver/btcpayserver/issues/4275#issuecomment-1308843465))

## Full animated flow (WIP - Will be updating)
![CheckoutFlowV0 2](https://user-images.githubusercontent.com/6250771/189557820-23890bcf-49aa-48e4-930d-80935a344bae.gif)

### Store Settings (WIP):
Whitelabeling
![Store Settings - General](https://user-images.githubusercontent.com/6250771/189557256-db2932ea-afa8-4645-ae0e-7e8d9b791772.png)

### Crowdfund Admin (WIP):
Addition of the custom metadata / form builder
The form builder would be encapsulated and modular so it can be leveraged in all plugins (Crowdfund, etc..), as each application might require different metadata.
![Apps - Point of Sale](https://user-images.githubusercontent.com/6250771/189557266-7b57559e-deab-42b1-8764-bb668fa60dc0.png)

### Point of Sale
![POS + Cart](https://user-images.githubusercontent.com/6250771/189557351-9a028231-8b94-4a49-9f3a-6eab2a059972.png)

### Cart + Review
![Cart](https://user-images.githubusercontent.com/6250771/189557588-ab565e67-acea-448c-9937-702ea626cd41.png)

### Metadata (WIP):
![Metadata](https://user-images.githubusercontent.com/6250771/189557615-1c2f90d4-eea5-4466-8685-88c1f3569aab.png)

### Invoice
![Invoice](https://user-images.githubusercontent.com/6250771/189557698-b16b073a-1b56-4ddf-bc8b-529ee25da978.png)

### Payment Success
![Payment Complete](https://user-images.githubusercontent.com/6250771/189557707-52131d01-bc3e-464e-84b5-9f19b4d42c08.png)

### Receipt
![Receipt](https://user-images.githubusercontent.com/6250771/189557717-150b70d5-3104-41f8-b968-053664d5c5ef.png)

**Comment by NicolasDorier** [comment]
The PR says "Improve point of sale", but it seems to be wider than point of sale.
PoS refers only to the "Tea shop" page.

Checkout refactoring isn't part of PoS.

Not anything wrong to put all in the same issue, but I was surprised by the title.

**Comment by dstrukt** [comment]
That's definitely fair, struggled with the title, as you point out, it's a subset of the POS, and the entire flow includes quite a few views and modifications .. will try think on a better name, but just wanted to capture everything in one Issue with related issues so we have a clearly defined path while we work through the sub-issues.

Also open to how we might want to deal with these larger initiatives over GH in the future as well, cc @pavlenex

**Comment by pavlenex** [comment]
@dennisreimann and I talked, it seems tackling this will be a huge overtaking that we yet have to fully flash out and define. I think that for 1.8.0 we can scope the PoS to just improving the keypad view #3668 and see how to further improve it in 1.9.0. Would require consensus between @dennisreimann and @dstrukt .

**Comment by dennisreimann** [comment]
We can continue here once #4568 is merged. The remaining tasks will be:

- [ ] Updated Point of Sale + Cart view
- [ ] POS: Make it clearer, how the custom amount works ([details](https://github.com/btcpayserver/btcpayserver/issues/4275#issuecomment-1308843465))

**Comment by pavlenex** [comment]
@dstrukt can we close this one and open a new issue specifically tracking the two remaining items on the list, that are clearly scoped?

**Comment by dstrukt** [comment]
Going to do as @dennisreimann and @pavlenex have suggested and isolate new Issues for the specific POS views. The scope of this ticket has been addressed.

Closing!
